### ISSUE TITLE
LimeSuiteNG on arm does not recognize LimeSDR Mini

### ISSUE THREAD

**Issue by davecrump** [myriadrf/LimeSuiteNG#91]
Using a Raspberry Pi 4 and the latest (fully updated) version of Bookworm Lite. Today I updated from LimeSuiteNG commit https://github.com/myriadrf/LimeSuiteNG/commit/626a781f2f1d20e25b3e731f3fe8fb350a7adc97 to commit [22f2b2e](https://github.com/myriadrf/LimeSuiteNG/commit/22f2b2e8095990a4ff3f556beb8c3c456632ca76).  I know that many changes have been implemented over this period!

After this LimeSuiteNG update, LimeSDR Minis (version 1 or version 2) are no longer recognised by LimeSuiteNG, but are recognised by LimeSuite.

```
pi@raspberrypi:~ $ lsusb
Bus 002 Device 002: ID 0403:601f Future Technology Devices International, Ltd FT601 32-bit FIFO IC
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
pi@raspberrypi:~ $ limeDevice --full
No devices found
pi@raspberrypi:~ $ LimeUtil --make
Make device
  Device name: LimeSDR-Mini
  Expansion name: UNSUPPORTED
  Firmware version: 6
  Hardware version: 2
  Protocol version: 1
  Gateware version: 1
  Gateware revision: 30
  Gateware target: LimeSDR-Mini
  Serial number: 0x1d4c2cae8c28a4
  Free connection... OK
```

LimeSDR Minis were recognised by commit https://github.com/myriadrf/LimeSuiteNG/commit/626a781f2f1d20e25b3e731f3fe8fb350a7adc97.

Of course, calls to LMS_GetDeviceList(NULL) return 0 devices making LimeSuiteNG unusable with a LimeSDR Mini.

Thanks

Dave, G8GKQ

**Comment by rjonaitis** [comment]
Works fine for me on RPi4, was libusb-1.0 dev installed when you compiled LimeSuiteNG?

when you run cmake check the enabled features:
```
- Found libusb-1.0:
--  - LIBUSB_1_INCLUDE_DIRS: /usr/include/libusb-1.0
--  - LIBUSB_1_LIBRARIES: /usr/lib/aarch64-linux-gnu/libusb-1.0.so

-- ######################################################
-- ## LimeSuiteNG enabled features
-- ######################################################
-- 
 * USB_FTDI, USB support for FTDI
```

**Comment by davecrump** [comment]
Thanks Ricardas

Yes - all looks OK:

```
-- Found Threads: TRUE
-- Found libusb-1.0:
--  - LIBUSB_1_INCLUDE_DIRS: /usr/include/libusb-1.0
--  - LIBUSB_1_LIBRARIES: /usr/lib/aarch64-linux-gnu/libusb-1.0.so
-- Could NOT find Gnuplot (missing: GNUPLOT_EXECUTABLE)
-- Found wxWidgets: -L/usr/lib/aarch64-linux-gnu;-pthread;;;-lwx_gtk3u_richtext-3.2;-lwx_gtk3u_core-3.2;-lwx_baseu-3.2;-lwx_gtk3u_gl-3.2 (found version "3.2.2")
-- wxWidgets_DEFINITIONS: _FILE_OFFSET_BITS=64;WXUSINGDLL;__WXGTK__
-- wxWidgets_INCLUDE_DIRS: /usr/lib/aarch64-linux-gnu/wx/include/gtk3-unicode-3.2;/usr/include/wx-3.2
-- wxWidgets_LIBRARIES: -L/usr/lib/aarch64-linux-gnu;-pthread;;;-lwx_gtk3u_richtext-3.2;-lwx_gtk3u_core-3.2;-lwx_baseu-3.2;-lwx_gtk3u_gl-3.2
-- Found OpenGL: /usr/lib/aarch64-linux-gnu/libOpenGL.so
-- OPENGL_INCLUDE_DIR: /usr/include
-- OPENGL_LIBRARIES: /usr/lib/aarch64-linux-gnu/libOpenGL.so;/usr/lib/aarch64-linux-gnu/libGLX.so;/usr/lib/aarch64-linux-gnu/libGLU.so
-- Enabling native optimizations
-- Found Git: /usr/bin/git (found version "2.39.2")
-- Module Soapy_limesuiteng configured with version: 0.2.0-22f2b2e8
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
--
-- ######################################################
-- Packages found:
 * Threads
 * libusb-1.0
   Adds USB communication support
 * wxWidgets
 * OpenGL
 * SoapySDR
   Adds LimeSuiteNG bindings for SoapySDR
 * Git

-- ######################################################
-- Missing required packages:
 * LinuxKernelHeaders

-- ######################################################
-- Missing recommended packages:

-- ######################################################
-- Missing optional packages:
 * Gnuplot
 * Doxygen

-- ######################################################
-- ## LimeSuiteNG enabled features
-- ######################################################
--
 * HEADERS, The limesuiteng headers
 * LIBRARY, The limesuiteng library
 * USB_FTDI, USB support for FTDI
 * USB_FX3, USB support for Cypress FX3
 * LIMEPCIE, PCIe support
 * LIMEPCIE_KERNEL_MODULE, Build Linux LimePCIe kernel module
 * EXAMPLES, LimeSuite library API examples
 * CLI, LimeSuite command line interface tools
 * GUI, GUI Application for LimeSuite
 * AMARISOFT_PLUGIN, LimeSuite Amarisoft integration plugin
 * DESKTOP, LimeSuiteNG freedesktop integration
 * SOAPYSDR, SoapySDR bindings for LimeSuiteNG
 * UDEV_RULES, Install Linux udev rules

-- ######################################################
-- ## LimeSuiteNG disabled features
-- ######################################################
--
 * DKMS, DKMS support for lime PCIe kernel module
 * GNURADIO, GNU Radio 3.9+ Plug-in
 * DOXYGEN, Build Doxygen documentation

-- Install prefix: /usr/local
-- Build timestamp: 2024-08-06
-- Lime Suite version: 0.2.0-g22f2b2e8
-- ABI/so version: 0.2-1
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/LimeSuiteNG/build
```
Although I did get this warning:
```
-- Enabling native optimizations
CMake Warning (dev) at /usr/share/cmake-3.25/Modules/FetchContent.cmake:1279 (message):
  The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
  not set.  The policy's OLD behavior will be used.  When using a URL
  download, the timestamps of extracted files should preferably be that of
  the time of extraction, otherwise code that depends on the extracted
  contents might not be rebuilt if the URL changes.  The OLD behavior
  preserves the timestamps from the archive instead, but this is usually not
  what you want.  Update your project to the NEW behavior or specify the
  DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
  robustness issue.
Call Stack (most recent call first):
  external/CMakeLists.txt:11 (fetchcontent_declare)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Building KissFFT with datatype: float

```
I'll try reverting back to the earlier commit tommorrow and let you know how I get on.

Dave

**Comment by rjonaitis** [comment]
That warning is due to CMake version, you can ignore it.

Just to make sure try to run `limeDevice` from the build/bin directory, as the global installed binary might be from different version.

**Comment by davecrump** [comment]
This issue is reproducable.

Build a fresh install of raspios bookworm arm64 lite, and install the LimeSuiteNG dependencies.  Then git clone the latest version of LimeSuiteNG, cmake, make and install.  Connect a LimeSDR Mini; limeDevice --full will then return "No devices found".

If you then delete the LimeSuiteNG folder, git clone the latest version of LimeSuite and then checkout an older build (I used commit 5afb2a3 ) and cmake, make and install.  limeDevice --full then sees the LimeSDR Mini.

If you delete the LimeSuiteNG folder again, git clone the latest version of LimeSuiteNG, cmake, make and install.  Connect a LimeSDR Mini; limeDevice --full will then return "No devices found".

Next, again delete the LimeSuiteNG folder, git clone the latest version of LimeSuite and then checkout the older build (5afb2a3) and cmake, make and install.  limeDevice --full then sees the LimeSDR Mini.

Next, DO NOT delete the LimeSuiteNG folder, but just checkout the latest commit (I used 770847c) and then cmake, make and install.   limeDevice --full then sees the LimeSDR Mini.

So it appears that there is something in the older builds of LimeSuiteNG that is required to still be in place for the newer builds of LimeSuiteNG to work correctly.  I hope that gives you enough to reproduce the issue and track it down.

I had been deleting old builds of LimeSuiteNG on each update, which is why I get the issue and you had not seen it.

Dave
